docs/migration_guide.md

```diff
@@ -1,36 +1,28 @@
-# Harbor upgrade and migration guide
+# Harbor Upgrade and Migration Guide
 
-This guide only covers upgrade and migration to version >= v1.8.0
+This guide covers upgrade and migration to versions >= v1.9.0. This guide only covers migration from v1.7.x and later to the current version. If you are upgrading from an earlier version, refer to the migration guide in the `release-1.7.0` branch to upgrade to v1.7.x first, then follow this guide to perform the migration to this version. 
 
-When upgrading your existing Harbor instance to a newer version, you may need to migrate the data in your database and the settings in `harbor.cfg`. 
-Since the migration may alter the database schema and the settings of `harbor.cfg`, you should **always** back up your data before any migration.
+When upgrading an existing Harbor 1.7.x instance to a newer version, you might need to migrate the data in your database and the settings in `harbor.cfg`. 
+Since the migration might alter the database schema and the settings of `harbor.cfg`, you should **always** back up your data before any migration.
 
-**NOTE:**
+**NOTES:**
 
 - Again, you must back up your data before any data migration.
+- Since v1.8.0, the configuration of Harbor has changed to a `.yml` file. If you are upgrading from 1.7.x, the migrator will transform the configuration file from `harbor.cfg` to `harbor.yml`. The command will be a little different to perform this migration, so make sure you follow the steps below.
+- In version 1.9.0, some containers are started by `non-root`. This does not pose problems if you are upgrading an officially released version of Harbor, but if you have deployed a customized instance of Harbor, you might encounter permission issues.
+- In previous releases, user roles took precedence over group roles in a project. In this version, if a user roles and group roles are combined so that the user has both the user and group role. This might cause the roles of certain users to change during upgrade.
```

Comment:
maybe it's more accurate to say that the user has the higher permission set of the two.  I believe this is true even if user rights are of higher permissions than group rights